### ISSUE TITLE
colour space confusion on macOS

### ISSUE THREAD

**Issue by rinkjames** [kovidgoyal/kitty#3692]
I have read this closed issue #2249, however I'm still a little confused about how to get colours to display accurately in my kitty terminal. I may not completely understand colour spaces and colour profiles... some help would be super appreciated!

Basically, I'm using a theme conf file in my $HOME/.config/kitty/themes folder but just to test colours I've been using e.g. `kitty --config=NONE -o background=#61afef`. Using macOS' digital colour meter, the RRGGBB colour shown is accurate when displaying native values (first image below), _but_ when I measure #61afef elsewhere incl my web browser (firefox via [google](https://www.google.com/search?client=firefox-b-d&q=%2361afef)) and my text editor (atom via [pigments package](https://github.com/abe33/atom-pigments)), the native value is totally different and the sRGB is very close but off by 1 or 2 values (I assume that's variation due to my device's unique colour profile?). See below images.

How do I get the colours across my applications to match as much as possible? I suppose I could just use the native values displayed in the other applications but those values don't seem to be consistent. Is there not an _accurate_ conversion between native values and sRGB so I can 'hack' my kitty theme to display the correct RRGGBB values in sRGB? i.e. background=#RRGGBB in kitty.conf displays as #61afef sRGB in kitty terminal?

| kitty | firerox | atom editor |
| :---:         |     :---:      |          :---: |
| <img width="761" alt="kitty" src="https://user-images.githubusercontent.com/13288151/120921140-958e0600-c6c2-11eb-8eca-83467a7a89e4.png">   | <img width="755" alt="firefox" src="https://user-images.githubusercontent.com/13288151/120921141-96bf3300-c6c2-11eb-9db0-f9b2ab6e79e7.png">    | <img width="762" alt="atom" src="https://user-images.githubusercontent.com/13288151/120921142-9757c980-c6c2-11eb-847b-a80c278c6d7a.png">   |

**Comment by Luflosi** [comment]
Color is very complicated. Every time I dive into the rabbit hole, I feel like I understand even less and have more questions than before.

**Comment by kovidgoyal** [comment]
Not that I know of, basically you need to gamma correct RGB colors based
on your monitors output profile. This varies from monitor to monitor.
kitty does no gamma correction, there is an open PR to add some, cant
remember the number but should be easy to find. Gamma correction
involves a parameter that can take basically arbitrary values. There is
no "correct" or "universal" value for this. So different applications do
different things.

Bottom line: You will need to manually adjust colors in kitty.conf if
you want it to match something else. If you can find out what gamma
factor that something else uses, it should be easy to cook up a script
to adjust the color values kitty uses in the same fashion.

Closing as there isn't really anything actionable here, but feel free to
discuss some more.

**Comment by rinkjames** [comment]
After a fair bit of reading on colour spaces and colour profile conversions, I've concluded that due to the way macOS displays colours and makes colour profile conversions with its Digital Colour Meter, I'm just going to have to live with the 'best fit' type of solution.

[This article](https://www.ricciadams.com/articles/osx-color-conversions/#errors) was especially helpful for understanding how colour profile conversion errors in macOS _probably_ mean it's impossible to have a hex colour code in kitty.conf that always converts perfectly using my display's unique colour profile to a desired sRGB hex colour code. So for my external monitor's colour profile, #74ADEA is converted to #62AFEF which is probbaly as close as I can get to my desired colour, #61AFEF. Some other colours seem to convert perfectly, but clearly there're some colour ranges that will be prone to rounding errors.

^ I say probably, because colour apparently is quite complicated. Maybe there's a way...
